packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -499,10 +497,47 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
             }
         }
 
+        if (!this.odspSummaryUploadManager) {
+            this.odspSummaryUploadManager = await this.summaryModuleP
+            .then(async (m) => {
+                this.odspSummaryModuleLoaded = true;
```

Comment:
I'm not sure having this.odspSummaryModuleLoaded gives us much - I'd remove it.
I'd also not reset this.summaryModulePon failure - if we wailed to load it, it's very unlikely it will succeed on repeat - I'd treat it as catastrophic event.
That said, I'm not sure how it all works in offline - does load fail? If so, it feels like retrying should be baked somehow here (i.e. not assume that summarizer will retry). Does webpack inject code to do proper retries in such cases?
Same concerns (on failures due to network issues) apply to your previous change as well - let's do proper due diligence to understand landscape and what is expected from us.

Reply:
It gives us 3-4 KB of saving which is about 3-4% now. @christiango @aboktor. Let me know if you feel this is significant enough to have this dynamically loaded.

Reply:
Not sure I understand the question here, if this code indeed isn't needed on boot then yeah we should delay load it especially since it will likely grow over time.

In terms of failure, any script download can fail. The Loop team customizes the webpack script loader to retry 3 times, but not all consumers of Fluid Framework will do this. That being said, that's a problem with the script download already and not new to this PR because of delay loading.